storage/innobase/btr/btr0cur.cc

```diff
@@ -4370,7 +4370,7 @@ btr_cur_pessimistic_update(
 					 cursor, offsets, offsets_heap,
 					 new_entry, &rec,
 					 &dummy_big_rec, n_ext, NULL, mtr);
-	ut_a(err == DB_SUCCESS);
+	if (err) return err;
```

Comment:
Can you add a comment that explains when exactly this can happen? Something like the following, perhaps?
```c++
	if (UNIV_UNLIKELY(err != DB_SUCCESS)) {
		/* This should only fail when the log for online ALTER TABLE cannot be written. */
		ut_ad(dict_index_is_online_ddl(index));
		ut_ad(err == DB_TEMP_FILE_WRITE_FAIL);
		return err;
	}
```
I can hardly imagine that the scenario is like that, because I would expect that log to be written at transaction commit time, not here. Maybe we are running out of space when trying to extend the tablespace? If there is a specific error code for that, please add a debug assertion to check for it.